static/js/lib/ckeditor/plugins/LegacyShortcodes.ts

```diff
@@ -0,0 +1,194 @@
+import { ShowdownExtension } from "showdown"
+import Plugin from "@ckeditor/ckeditor5-core/src/plugin"
+import { editor } from "@ckeditor/ckeditor5-core"
+
+import MarkdownSyntaxPlugin from "./MarkdownSyntaxPlugin"
+import { TurndownRule } from "../../../types/ckeditor_markdown"
+import { LEGACY_SHORTCODES } from "./constants"
+
+const shortcodeClass = (shortcode: string) => `legacy-shortcode-${shortcode}`
+
+const DATA_ISCLOSING = "data-isclosing"
+const DATA_ARGUMENTS = "data-arguments"
+
+class LegacyShortcodeSyntax extends MarkdownSyntaxPlugin {
+  static get pluginName(): string {
+    return "LegacyShortcodeSyntax"
+  }
+
+  get showdownExtension() {
+    return function legacyShortcodeExtension(): ShowdownExtension[] {
+      return LEGACY_SHORTCODES.map(
+        (shortcode: string): ShowdownExtension => {
+          const shortcodeRegex = new RegExp(`{{< /?${shortcode} (.*?)>}}`, "g")
+          const closingShortcodeRegex = new RegExp(
+            `{{< /${shortcode} (.*?)>}}`,
+            "g"
+          )
+
+          return {
+            type:    "lang",
+            regex:   shortcodeRegex,
+            replace: (stringMatch: string, shortcodeArgs: string | null) => {
+              const isClosing = JSON.stringify(
+                closingShortcodeRegex.test(stringMatch)
+              )
+
+              const tag = `<span ${DATA_ISCLOSING}="${isClosing}" ${
+                shortcodeArgs ?
+                  `${DATA_ARGUMENTS}="${escape(shortcodeArgs)}"` :
```

Comment:
**Question/Suggestion**: I was unfamiliar with `escape` and VSCode highlighted it, so I wlooked it up on MDN. MDN says not to use it, but doesn't really explain why. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape Would `encodeURIComponent` work instead?

Reply:
I think we can use encoreURIComponent, I'll swap that in and make sure the tests pass real quick. The reason to do it is just that if it's not escaped then the text w/in the arguments to a shortcode can cause issues correctly finding the boundaries of the HTML tags later on and things get all messed up. When I was hacking on it I actually used `btoa` and `atob` to be 100% sure I wasn't going to get any characters I didn't expect in the arguments string, haha

Reply:
yep, I already made the change since all the tests pass with it